pkg/controller/discovery/web/web.go

```diff
@@ -359,7 +359,7 @@ func (h *BaseHandler) allow(sar auth.SelfSubjectAccessReview) int {
 	post := restClient.Post()
 	post.Resource("selfsubjectaccessreviews")
 	post.Body(&sar)
-	reply := post.Do()
+	reply := post.Do(context.Background())
```

Comment:
context.TODO() might the more appropriate call here because we have no intention of doing anything with that context at the moment? Thoughts?

Reply:
I can do that. Previously, I got comments when I'd used TODO() saying that Background() was more appropriate when we had no intention of changing it later, and TODO() was more appropriate when we weren't sure. @alaypatel07 -- amy thoughts here?

Reply:
hmm yeah, I think the way I learnt to use contexts is, if the context will ever be used with something higher level, I use TODO, and if I know I will never touch the context again, I use Background. That way all the TODOs in the codebase refer to contexts that will eventually get meaning and the rest will be Background.  If that is the wrong way of thinking please I know I am curious to learn. 

and yeah I was the one to share and tell this in the team in reviews :) sorry if it was mid-directed.

Reply:
So the question here is what comes next @alaypatel07 is correct:

https://golang.org/pkg/context/#TODO
> Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

and https://golang.org/pkg/context/#Background

> It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

My thinking is that eventually, we would want to handle contexts in this code path but it still does not feel like a top level context to me so IDK. 

I can be persuaded, but I think we probably want to eventually have some context to the HTTP call?